src/zenml/config/pipeline_configurations.py

```diff
@@ -86,3 +89,23 @@ def docker_settings(self) -> "DockerSettings":
             DOCKER_SETTINGS_KEY, {}
         )
         return DockerSettings.parse_obj(model_or_dict)
+
+    @contextmanager
+    def in_design_time(self) -> Iterator[None]:
+        """Context manager to mark a pipeline as being in design time.
```

Comment:
What does something being 'in design time' mean? Maybe add something to the docstring on this?

Reply:
I read a bit and it's 'a thing'! Is this what you also have elsewhere referred to as 'composition'?

Reply:
Yep, design time is a composition in my terms. Shall I unify it all to composition/execution instead of design/run times?

Reply:
I don't mind really, but yeah it should be one word. I heard them both first this week so not even sure it's a thing in ZenML docs?

Reply:
Seems that `compile` is the term used internally for design time. Will rename

Reply:
Renamed